flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/runner/DispatcherRunnerImpl.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.dispatcher.runner;
+
+import org.apache.flink.runtime.concurrent.FutureUtils;
+import org.apache.flink.runtime.dispatcher.Dispatcher;
+import org.apache.flink.runtime.dispatcher.DispatcherFactory;
+import org.apache.flink.runtime.dispatcher.PartialDispatcherServices;
+import org.apache.flink.runtime.rpc.RpcService;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Runner responsible for executing a {@link Dispatcher} or a subclass thereof.
+ */
+public class DispatcherRunnerImpl<T extends Dispatcher> implements DispatcherRunner {
+
+	private final T dispatcher;
+
+	DispatcherRunnerImpl(
+			DispatcherFactory<T> dispatcherFactory,
+			RpcService rpcService,
+			PartialDispatcherServices partialDispatcherServices) throws Exception {
+		this.dispatcher = dispatcherFactory.createDispatcher(
+			rpcService,
+			partialDispatcherServices);
+		dispatcher.start();
+	}
+
+	@Override
+	public T getDispatcher() {
+		return dispatcher;
+	}
+
+	@Override
+	public CompletableFuture<Void> closeAsync() {
+		Collection<CompletableFuture<Void>> terminationFutures = new ArrayList<>(2);
+
+		terminationFutures.add(dispatcher.closeAsync());
+
+		return FutureUtils.completeAll(terminationFutures);
```

Comment:
There is only one future inside `terminationFutures` here.

Reply:
Yes indeed. Will remove it.